cmd/gazelle: TestImportCollision is flaky

### What version of gazelle are you using?

33ccc50dcd2d0b9d8937bc8497fa13505bd7a665

### What version of rules_go are you using?

v0.20.3

### What version of Bazel are you using?

2.0.0

### Does this issue reproduce with the latest releases of all the above?

n/a

### What operating system and processor architecture are you using?

darwin amd64

### What did you do?

bazel test --runs_per_test=20 --test_filter=TestImportCollision //cmd/gazelle:go_default_test

### What did you expect to see?

success

### What did you see instead?

```
--- FAIL: TestImportCollision (2.13s)
    integration_test.go:2918: want imports github.com/Selvatico/go-mocket and github.com/selvatico/go-mocket resolve to the same repository rule name com_github_selvatico_go_mocket, got imports github.com/selvatico/go-mocket and github.com/Selvatico/go-mocket resolve to the same repository rule name com_github_selvatico_go_mocket
```

The two repo names are occasionally out of order (they differ in capitalization). The order of generated rules seems to be non-deterministic.

--- a/language/go/update.go
+++ b/language/go/update.go
@@ -17,6 +17,7 @@ package golang
 
 import (
 	"path/filepath"
+	"sort"
 	"strings"
 
 	"github.com/bazelbuild/bazel-gazelle/language"
@@ -83,6 +84,7 @@ func (*goLang) ImportRepos(args language.ImportReposArgs) language.ImportReposRe
 			}
 		}
 	}
+	sortRules(res.Gen)
 	return res
 }
 
@@ -106,4 +108,13 @@ func setBuildAttrs(gc *goConfig, r *rule.Rule) {
 		extraArgs := strings.Split(gc.buildExtraArgsAttr, ",")
 		r.SetAttr("build_extra_args", extraArgs)
 	}
+}
+
+func sortRules(rules []*rule.Rule) {
+	sort.SliceStable(rules, func(i, j int) bool {
+		if cmp := strings.Compare(rules[i].Name(), rules[j].Name()); cmp != 0 {
+			return cmp < 0
+		}
+		return rules[i].AttrString("importpath") < rules[j].AttrString("importpath")
+	})
 }

